src/test/unit/serial_msp_unittest.cc

```diff
@@ -0,0 +1,598 @@
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdint.h>
+#include <string.h>
+#include <math.h>
+
+extern "C" {
+    #include "platform.h"
+    #include "build_config.h"
+    #include "version.h"
+    #include "debug.h"
+    #include "common/axis.h"
+    #include "common/color.h"
+    #include "common/maths.h"
+
+    #include "drivers/system.h"
+    #include "drivers/sensor.h"
+    #include "drivers/accgyro.h"
+    #include "drivers/compass.h"
+    #include "drivers/serial.h"
+    #include "drivers/bus_i2c.h"
+    #include "drivers/gpio.h"
+    #include "drivers/timer.h"
+    #include "drivers/pwm_rx.h"
+
+    #include "rx/rx.h"
+
+    #include "io/rc_controls.h"
+    #include "io/gps.h"
+    #include "io/gimbal.h"
+    #include "io/ledstrip.h"
+    #include "io/serial_msp.h"
+    #include "io/escservo.h"
+
+    #include "telemetry/telemetry.h"
+
+    #include "sensors/sensors.h"
+    #include "sensors/boardalignment.h"
+    #include "sensors/sensors.h"
+    #include "sensors/battery.h"
+    #include "sensors/sonar.h"
+    #include "sensors/acceleration.h"
+    #include "sensors/barometer.h"
+    #include "sensors/compass.h"
+    #include "sensors/gyro.h"
+
+    #include "flight/mixer.h"
+    #include "flight/pid.h"
+    #include "flight/navigation.h"
+    #include "flight/imu.h"
+    #include "flight/failsafe.h"
+
+    #include "config/runtime_config.h"
+    #include "config/config.h"
+    #include "config/config_profile.h"
+    #include "config/config_master.h"
+}
+
+#include "unittest_macros.h"
+#include "gtest/gtest.h"
+
+#define MSP_PROTOCOL_VERSION                0
+
+#define API_VERSION_MAJOR                   1 // increment when major changes are made
+#define API_VERSION_MINOR                   15 // increment when any change is made, reset to zero when major changes are released after changing API_VERSION_MAJOR
+
+#define FLIGHT_CONTROLLER_IDENTIFIER_LENGTH 4
+#define FLIGHT_CONTROLLER_VERSION_LENGTH    3
+
+#define MSP_API_VERSION                 1    //out message
+#define MSP_FC_VARIANT                  2    //out message
+#define MSP_FC_VERSION                  3    //out message
+
+#define MSP_PID_CONTROLLER              59
+#define MSP_SET_PID_CONTROLLER          60
+
+#define MSP_PID                  112    //out message         P I D coeff (9 are used currently)
+#define MSP_PIDNAMES             117    //out message         the PID names
+#define MSP_SET_PID              202    //in message          P I D coeff (9 are used currently)
+
+#define INBUF_SIZE 64
```

Comment:
prefix each of these `#defines` with `TEST_`.  This will increase test robustness and be less confusing.